### PR TITLE
feat: make composer optional

### PR DESCRIPTION
**Pull request by shyim**
Allows disabling composer and adding own using packages

### CODE REVIEW

**Comment by domenkozar** [comment]
Same as #315, this should be called `languages.php.packages.composer` that can be set to `null`.

**Comment by kkirsche** [comment]
It sounds like that would be this piece of configuration from Rust:
https://github.com/cachix/devenv/blob/main/src/modules/languages/rust.nix#L31-L36 if I understood @domenkozar's comment in #315 and the languages file structure properly

File: src/modules/languages/php.nix [hunk 1]
```diff
@@ -162,6 +162,22 @@ in
       '';
     };
 
+    packages = lib.mkOption {
+      type = lib.types.submodule ({
+        options = {
+          composer = lib.mkOption {
+            type = lib.types.nullOr lib.types.str;
+            default = "composer";
```
Comment (domenkozar) [inline src/modules/languages/php.nix:170]: the type should be `types.package` and the default `cfg.package.packages.composer`.

Comment (shyim) [reply src/modules/languages/php.nix:170]: I have no idea how it should work with package 🤔 . I need to point to the `cfg.package.packages.XXXXX`

Comment (domenkozar) [reply src/modules/languages/php.nix:170]: This should work:

```nix
config = {
...
  languages.php.composer = lib.mkDefault cfg.pacakge.packages.composer;
...
}
```

**Comment by shyim** [comment]
@domenkozar refactored it like you said and works :)

**Comment by shyim** [comment]
macos fails randomly installing nix https://github.com/cachix/devenv/actions/runs/4102872885/jobs/7078523073 🙈

**Pull request merged**